packages/expect/__typetests__/expect.test.ts

```diff
@@ -111,141 +27,262 @@ declare module 'expect' {
+
+        function anotherTester(a, b, customTesters) {
+          expect(a).type.toBeAny();
+          expect(b).type.toBeAny();
+          expect(customTesters).type.toEqual<Array<Tester>>();
+          expect(this).type.toEqual<TesterContext>();
+          expect(this.equals).type.toEqual<EqualsFunction>();
+
+          return undefined;
+        },
+      ]),
+    ).type.toBeVoid();
+  });
+
+  test('.extend()', () => {
+    type MatcherUtils = typeof jestMatcherUtils & {
+      iterableEquality: Tester;
+      subsetEquality: Tester;
     };
-  };
+
+    expect(
+      _expect.extend({
+        // TODO `actual` should be allowed to have only `unknown` type
+        toBeWithinRange(actual: number, floor: number, ceiling: number) {
+          expect(this.assertionCalls).type.toBeNumber();
+          expect(this.currentTestName).type.toEqual<string | undefined>();
+          expect(this.customTesters).type.toEqual<Array<Tester>>();
+          expect(this.dontThrow).type.toEqual<() => void>();
+          expect(this.error).type.toEqual<Error | undefined>();
+          expect(this.equals).type.toEqual<EqualsFunction>();
+          expect(this.expand).type.toEqual<boolean | undefined>();
+          expect(this.expectedAssertionsNumber).type.toEqual<number | null>();
+          expect(this.expectedAssertionsNumberError).type.toEqual<
+            Error | undefined
+          >();
+          expect(this.isExpectingAssertions).type.toBeBoolean();
+          expect(this.isExpectingAssertionsError).type.toEqual<
+            Error | undefined
+          >();
+          expect(this.isNot).type.toEqual<boolean | undefined>();
+          expect(this.numPassingAsserts).type.toBeNumber();
+          expect(this.promise).type.toEqual<string | undefined>();
+          expect(this.suppressedErrors).type.toEqual<Array<Error>>();
+          expect(this.testPath).type.toEqual<string | undefined>();
+          expect(this.utils).type.toEqual<MatcherUtils>();
+
+          const pass = actual >= floor && actual <= ceiling;
+          if (pass) {
+            return {
+              message: () =>
+                `expected ${actual} not to be within range ${floor} - ${ceiling}`,
+              pass: true,
+            };
+          } else {
+            return {
+              message: () =>
+                `expected ${actual} to be within range ${floor} - ${ceiling}`,
+              pass: false,
+            };
+          }
+        },
+      }),
+    ).type.toBeVoid();
+
+    expect(_expect(100).toBeWithinRange(90, 110)).type.toBeVoid();
+    expect(_expect(101).not.toBeWithinRange(0, 100)).type.toBeVoid();
+
+    expect(
+      _expect({apples: 6, bananas: 3}).toEqual({
+        apples: _expect.toBeWithinRange(1, 10),
+        bananas: _expect.not.toBeWithinRange(11, 20),
+      }),
+    ).type.toBeVoid();
+  });
+
+  test('does not define the `.toMatchSnapshot()` matcher', () => {
+    expect(_expect(null)).type.not.toHaveProperty('toMatchSnapshot');
```

Comment:
`jestExpect` would not look right here, or? This is `expectExpect`, but `expect(expectExpect` is somewhat too much.

Reply:
haha